src/bepasty/apis/rest.py

```diff
@@ -157,9 +158,17 @@ def post(self):
             # Get file name from Transaction-ID and open from Storage
             trans_id_s = request.headers.get(TRANSACTION_ID)
             trans_id_b = trans_id_s if isinstance(trans_id_s, bytes_type) else trans_id_s.encode()
-            name_b = base64.b64decode(trans_id_b)
+            try:
+                name_b = base64_b64decode(trans_id_b)
+            except (base64.binascii.Error, TypeError):
+                raise BadRequest(description='{} is incorrect'.format(TRANSACTION_ID))
             name = name_b if isinstance(name_b, str) else name_b.decode()
-            item = current_app.storage.openwrite(name)
+            try:
+                item = current_app.storage.openwrite(name)
+            except (OSError, IOError) as e:
+                if e.errno == errno.ENOENT:
+                    raise BadRequest(description='{} is incorrect'.format(TRANSACTION_ID))
```

Comment:
guess this should be something like "could not find storage item for transaction id"

Reply:
fixed